// together-ai/resources/audio/audio.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as TranscriptionsAPI from "./transcriptions.mjs";
import { Transcriptions } from "./transcriptions.mjs";
import * as TranslationsAPI from "./translations.mjs";
import { Translations } from "./translations.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
export class Audio extends APIResource {
    constructor() {
        super(...arguments);
        this.transcriptions = new TranscriptionsAPI.Transcriptions(this._client);
        this.translations = new TranslationsAPI.Translations(this._client);
    }
    create(body, options) {
        return this._client.post('/audio/speech', {
            body,
            ...options,
            headers: buildHeaders([{ Accept: 'application/octet-stream' }, options?.headers]),
            stream: body.stream ?? false,
            __binaryResponse: true,
        });
    }
}
Audio.Transcriptions = Transcriptions;
Audio.Translations = Translations;
